src/samplers/hwinfo/mod.rs

```diff
@@ -0,0 +1,108 @@
+use serde::Serialize;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::io::Result;
+use std::path::Path;
+use walkdir::DirEntry;
+use walkdir::WalkDir;
+
+mod cache;
+mod cpu;
+mod memory;
+mod net;
+mod node;
+
+use cache::*;
+use cpu::*;
+use memory::*;
+use net::*;
+use node::*;
+
+#[derive(Serialize)]
+pub struct Hwinfo {
+    caches: Vec<Vec<Cache>>,
+    cpus: Vec<Cpu>,
+    memory: Memory,
+    network: Vec<Interface>,
+    nodes: Vec<Node>,
+}
+
+impl Hwinfo {
```

Comment:
My preference is to consolidate all the `Hwinfo` sources, e.g. `/proc/cpuinfo`, `proc/meminfo`, `/sys/devices/system/cpu/*` in this top level file. This way someone who is familiar with system info but not rezolus can quickly get an overall impression of what kind of info is being consumed by this module, and anybody can `cat` those paths to get examples of the data from a live system.

Reply:
I agree. This particular sampler is not well-structured. Perhaps we should have some common reader functions and intermediate representations of the data coming from these files - there may be overlap with some of the files used for gathering telemetry too.

Reply:
good idea. Cutting an issue (#11) and we can skip it for now.